components/SearchSuggestion/SearchAutosuggest.tsx

```diff
@@ -16,51 +16,120 @@ import { faGrid2 } from "@fortawesome/pro-solid-svg-icons";
 import { toTitleCase } from "~/config/utils/string";
 import { formatNumber } from "~/config/utils/number";
 import { highlightTextInSuggestion } from "./lib/util";
+import { useEffect, useState } from "react";
+import { faSearch } from "@fortawesome/pro-light-svg-icons";
 
 interface SearchSuggestionProps {
   suggestions: Suggestion[];
-  textToHighlight?: string;
-  handleSuggestionClick: (suggestion: Suggestion) => void;
+  searchString?: string;
+  handleSuggestionSelect: (suggestion: Suggestion) => void;
+  handleAllResultsSelect: () => void;
+  handleClose: () => void;
 }
 
 const SearchSuggestions = ({
   suggestions,
-  textToHighlight,
-  handleSuggestionClick,
+  searchString,
+  handleSuggestionSelect,
+  handleAllResultsSelect,
+  handleClose,
 }: SearchSuggestionProps) => {
+  const [selectedSuggestionIndex, setSelectedSuggestionIndex] =
+    useState<number>(0);
+
+  useEffect(() => {
+    const handleKeyUp = (event: KeyboardEvent) => {
+      if (event.key === "Enter") {
+        if (selectedSuggestionIndex === 0) {
+          handleAllResultsSelect();
+        } else {
+          handleSuggestionSelect(suggestions[selectedSuggestionIndex - 1]);
+        }
+      }
+      if (event.key === "Escape") {
+        handleClose();
+      } else if (event.key === "ArrowUp") {
+        if (selectedSuggestionIndex === -1 || selectedSuggestionIndex === 0) {
+          setSelectedSuggestionIndex(suggestions.length);
+        } else {
+          setSelectedSuggestionIndex(selectedSuggestionIndex - 1);
+        }
+      } else if (event.key === "ArrowDown") {
+        if (
+          selectedSuggestionIndex === -1 ||
+          selectedSuggestionIndex >= suggestions.length
+        ) {
+          setSelectedSuggestionIndex(0);
+        } else {
+          setSelectedSuggestionIndex(selectedSuggestionIndex + 1);
+        }
+      }
+    };
```

Comment:
Nit: Wondering if the use of switch case is more readable then nested if statements?
```suggestion
      switch (event.key) {
        case "Enter":
          if (selectedSuggestionIndex === 0) {
            handleAllResultsSelect();
          } else {
            handleSuggestionSelect(suggestions[selectedSuggestionIndex - 1]);
          }
          break;
        case "Escape":
          handleClose();
          break;
        case "ArrowUp":
          if (selectedSuggestionIndex === -1 || selectedSuggestionIndex === 0) {
            setSelectedSuggestionIndex(suggestions.length);
          } else {
            setSelectedSuggestionIndex(selectedSuggestionIndex - 1);
          }
          break;
        case "ArrowDown":
          if (
            selectedSuggestionIndex === -1 ||
            selectedSuggestionIndex >= suggestions.length
          ) {
            setSelectedSuggestionIndex(0);
          } else {
            setSelectedSuggestionIndex(selectedSuggestionIndex + 1);
          }
          break;
        default:
          break;
      }
```

Reply:
Good suggestion